components/Home/Achievements/index.tsx

```diff
@@ -0,0 +1,42 @@
+import { Title, Divider, Base64Image } from '../../Core'
+import { getShortDateOr } from '../../../lib/utils'
+import styles from './Achievements.module.css'
+import { DATE_NULL_REPLACEMENT } from '../../../lib/constants'
+import { Fragment } from 'react'
+
+interface AchievementProps {
+  amountToShow: number,
+  achievements: AchievementData[]
+}
+
+export default function Achievements({ amountToShow, achievements }: AchievementProps) {
+  const actualAmountToShow = Math.min(achievements.length, amountToShow)
+  const renderAchievements = achievements.slice(0, actualAmountToShow).map(a => (
```

Comment:
Using these sort of "renderAchievements" or similar "renderX" functions or variables begs the question could these be actually separated into Components? I personnally like keeping things quite simple in JSX side of a view and hide all the "uglyness" behind separate components. Then you can just map things out in JSX.

Reply:
It is a good point. Will mark it down for later.